### PR TITLE
[3.11] Pod Retention for Jenkins Pipeline Builds

### PR DESCRIPTION
**Pull request by adambkaplan**
Doc for the Pod Retention setting added to the Jenkins
Kubernetes plugin.

Bug [151940](https://bugzilla.redhat.com/show_bug.cgi?id=1515940)

See related PR jenkinsci/kubernetes-plugin#354

### CODE REVIEW

**Comment by adambkaplan** [comment]
/hold
Wait until approval upstream

File: using_images/other_images/jenkins.adoc
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: I would clarify that these build pods are Jenkins agent/slave pods created by kubernetes plugin.

**Review by gabemontero — CHANGES_REQUESTED** [review]
I would move this to using_images/other_images/jenkins_slave.adoc

**Comment by adambkaplan** [comment]
@gabemontero I'm a bit less inclined to move this section to the `jenkins_slave` page, since the latter goes into the details of our Maven and node.js slave images. Furthermore the behavior is governed by the kube plugin, which "lives" on the master.

**Comment by gabemontero** [comment]
@adambkaplan ... still mulling your rationale on the page location, but in the interim,  if we don't move the content, I think a link to your new section here on that page (perhaps in the beginning of the usage section) is helpful.  Your thoughts there?

**Comment by adambkaplan** [comment]
@gabemontero moved it over with an "intro note" on the jenkins page.

**Review by gabemontero — APPROVED** [review]
cool ... thanks @adambkaplan

**Comment by vikram-redhat** [comment]
@adambkaplan - is this ok to move forward with?

**Comment by adambkaplan** [comment]
@vikram-redhat this was merged upstream yesterday. Currently waiting on a new version of the plugin to be released, then we can add it to our Jenkins image.

**Comment by vikram-redhat** [comment]
Thanks @adambkaplan - ping the openshift docs team when ready.

File: using_images/other_images/jenkins.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: Are we still referring to these as Jenkins "Slaves" or can it be changed to "Agents"?

File: using_images/other_images/jenkins_slaves.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins_slaves.adoc]: s/via/by
s/"Pod Retention"/*Pod Retention*

File: using_images/other_images/jenkins_slaves.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins_slaves.adoc]: s/"..."/`...` (for all of the list items)
s/will keep/keeps
s/will use/uses

File: using_images/other_images/jenkins_slaves.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins_slaves.adoc]: s/Pod retention can also be overrode in the pipeline Jenkinsfile as follow:/You can override pod retention in the pipeline Jenkinsfile:

File: using_images/other_images/jenkins_slaves.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins_slaves.adoc]: We can remove the ==== here (puts a box around the code block)

File: using_images/other_images/jenkins_slaves.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins_slaves.adoc]: We can remove the ==== here (puts a box around the code block)

File: using_images/other_images/jenkins_slaves.adoc
Comment (bmcelvee) [inline using_images/other_images/jenkins_slaves.adoc]: ifdef::openshift-origin[]
For more information, see the link:https://github.com/jenkinsci/kubernetes-plugin[Kubernetes plug-in documentation].
endif::openshift-origin[]

**Comment by bmcelvee** [comment]
Just a few nits. Otherwise LGTM :stars: 

When it's ready we'll do QE review too.

File: using_images/other_images/jenkins.adoc
Comment (adambkaplan) [reply using_images/other_images/jenkins.adoc]: Looks like "Agents" is clearer - I can change it

**Comment by adambkaplan** [comment]
/hold cancel

per [openshift/jenkins#672](https://github.com/openshift/jenkins/pull/672#issuecomment-409673498) this is released.

**Comment by bmcelvee** [comment]
@xltian PTAL

**Comment by xltian** [comment]
@wzheng1 as the QE owner for this feature will review.

**Comment by bmcelvee** [comment]
Thanks, @xltian and @wzheng1!

**Comment by adambkaplan** [comment]
@wzheng1 should have updated the Trello card. This was completed Tuesday and should be available in the latest openshift Jenkins image.

**Comment by wzheng1** [comment]
Feature is still under testing and with bug blocked. But it look good to me just for this doc.

**Comment by bmcelvee** [comment]
Thanks, @bparees!

**Pull request merged**
